### ISSUE TITLE
Notification Email

### ISSUE THREAD

**Issue by fequesada** [nabeelio/phpvms#1676]
**Describe the solution you'd like**
Notification for change account status.

**Describe alternatives you've considered**
How registration approval notification emails are sent, it could also be sent an email for an account status change.

**Additional context**

**Comment by fequesada** [comment]
I talked about the suggestion because sometimes we have to warn or suspend a pilot for some reason, and there isn't that possibility. We would have to wait for the person to contact us, and then explain the reason for such suspension or warning. And sometimes, we want to send an email to specific pilots about something they did wrong or forgot. But it was just a suggestion because, in my virtual airline, we already have almost 50 active pilots. That's why I said it would make things a lot easier.

Safe flights!

**Comment by arthurpar06** [comment]
Hi,
What do you think about sending emails to specific pilots from the admin interface? Similar to creating a news, you enter the subject, the content, then you choose your recipients and phpVMS sends the email for you.

**Comment by fequesada** [comment]
No, if a pilot is warned, suspended, or banned, they receive an email notification. Captain X committed some infraction; we changed his status to suspended. He receives an email notifying him that he is suspended.

**Comment by fequesada** [comment]
That's it!

**Comment by arthurpar06** [comment]
Okay, but then my question is, what does it really achieve? 

Because a suspended pilot who tries to log in will see a message saying that they can't log in because they are suspended. If we send an email that just says "You are suspended," without explaining why the person will still contact you to ask for an explanation of their suspension.

**Comment by FatihKoz** [comment]
Nothing will be achieved, it will be a mail sent out. That's it.

What can be done is maybe utilizing a field for administrative notes, (like remarks) make it visible in that email and also make it visible on login attempts.

Then it "maybe" can be useful.

**Comment by fequesada** [comment]
> Nothing will be achieved, it will be a mail sent out. That's it.
> 
> What can be done is maybe utilizing a field for administrative notes, (like remarks) make it visible in that email and also make it visible on login attempts.
> 
> Then it "maybe" can be useful.

Exactly, this would explain the reason for the suspension/ban, providing clarity on the matter. More specific matters will be discussed via Discord. Failure to respond may result in suspension.

**Comment by FatihKoz** [comment]
User will be suspended when this happens. So your "failure to respond may result suspension" is not right :)

**Comment by FatihKoz** [comment]
Either way, user will not be able to login, and will contact staff.

So that email will be really useless and not operational at all 😉

**Comment by FatihKoz** [comment]
Another option, add a new state WARNING ... Put pilots to that state with a reason, if no reply in x days, auto suspend ;)

Complete process can be automated, like a user does not fly minimum amount per month, crashes too often etc. etc.

WARN them, display a message at dashboard with remaining days. Until suspension of clearance of warning.

**Comment by fequesada** [comment]
A suggestion, I'm not sure if that's what Fatih meant. We have the Management Notes field: Perhaps if every time we added something there, the pilot would be notified via email, in that case, we wouldn't need to make this whole change.

**Comment by fequesada** [comment]
> Another option, add a new state WARNING ... Put pilots to that state with a reason, if no reply in x days, auto suspend ;)
> 
> Complete process can be automated, like a user does not fly minimum amount per month, crashes too often etc. etc.
> 
> WARN them, display a message at dashboard with remaining days. Until suspension of clearance of warning.

A great idea as well!

**Comment by FatihKoz** [comment]
> A suggestion, I'm not sure if that's what Fatih meant. We have the Management Notes field: Perhaps if every time we added something there, the pilot would be notified via email, in that case, we wouldn't need to make this whole change.

This would be too tricky, checking updates, sending mails on each update is not logical :(

**Comment by arthurpar06** [comment]
> Another option, add a new state WARNING ... Put pilots to that state with a reason, if no reply in x days, auto suspend ;)
> 
> Complete process can be automated, like a user does not fly minimum amount per month, crashes too often etc. etc.
> 
> WARN them, display a message at dashboard with remaining days. Until suspension of clearance of warning.

Hmm, it requires thinking about conditions to warn someone, a warning time that may depend on the condition that triggered the warning, conditions that automatically remove the warning, etc... It's a bit more complex I think, but it could be a nice idea

**Comment by FatihKoz** [comment]
I can do it with a module :) like auto reject logic

**Comment by arthurpar06** [comment]
If warning is a UserState and we want to display a mesasge on the dahsboard like for on leave it will also require changes in the core

**Comment by FatihKoz** [comment]
Put the flash message and load dashboard, it will be visible... it is designed to show latest flash message, no need to send it from core.

A widget without visible items can do this easily, checks user warnings, push messages, keep track of them etc.

Anyway, it is another subject 😉
